tools/build_config/Jenkinsfile

```diff
@@ -60,57 +60,18 @@ def get_branch_name(String job_name) {
   }
 }
 
-properties([
-  parameters([
-    string(
-      defaultValue: get_branch_name(env.JOB_BASE_NAME),
-      description: 'The name of the branch to build.',
-      name: 'BRANCH_NAME',
-      trim: true
-    ),
-    string(
-      defaultValue: get_release_type(env.JOB_BASE_NAME),
-      description: 'The type of the build e.g. "nightly", "release", "pull_request".',
-      name: 'RELEASE_TYPE',
-      trim: true
-    ),
-    string(
-      defaultValue: get_agent(env.JOB_BASE_NAME),
-      description: 'The agent to execute the pipeline on.',
-      name: 'AGENT',
-      trim: true
-    ),
-    string(
-      defaultValue: get_matlab_release(env.JOB_BASE_NAME),
-      description: 'The release number of the Matlab to load e.g. R2019b.',
-      name: 'MATLAB_VERSION', trim: true
-    ),
-    string(
-      defaultValue: '3.19',
-      description: 'The version of CMake to run the build with.',
-      name: 'CMAKE_VERSION',
-      trim: true
-    ),
-    string(
-      defaultValue: '6.3.0',
-      description: 'The version of GCC to build with.',
-      name: 'GCC_VERSION',
-      trim: true
-    ),
-    string(
-      defaultValue: '2017',
-      description: 'The year of the version of Visual Studio to build with.',
-      name: 'VS_VERSION',
-      trim: true
-    ),
-    string(
-      defaultValue: '1.77',
-      description: 'The version of CppCheck tooling to load to provide the code-style checks.',
-      name: 'CPPCHECK_VERSION',
-      trim: true
-    )
-  ])
-])
+String get_param(String param_name, String default_val) {
```

Comment:
Checking that you have to move the parameters([ block after post-github_status. It means that the diff cannot compare parameters and show they are identical

Reply:
I moved it because it puts all the functions in one place. One of the future tasks (distant, I suspect) is to move all the shared functions into a Jenkins libray. This just makes that one bit easier. The parameters are identical, but the defaults have been removed from all except the three which are required Jenkins params.